tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,354 @@
+    else:
+        pytest.fail("Failed to find program '{}' in container '{}'"
+                    .format(program_name, container_name))
+
+    logger.info("Waiting until container '{}' is stopped...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS,
+                  CONTAINER_CHECK_INTERVAL_SECS,
+                  check_container_state, duthost, container_name, False),
+                  "Failed to stop container '{}'".format(container_name))
+    logger.info("Container '{}' was stopped".format(container_name))
+
+    logger.info("Waiting until container '{}' is restarted...".format(container_name))
+    restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                           CONTAINER_CHECK_INTERVAL_SECS,
+                           check_container_state, duthost, container_name, True)
+    if not restarted:
+        if is_hiting_start_limit(duthost, container_name):
+            logger.info("{} hits start limit and clear reset-failed flag".format(container_name))
+            duthost.shell("sudo systemctl reset-failed {}.service".format(container_name))
+            duthost.shell("sudo systemctl start {}.service".format(container_name))
+            pytest_assert(wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                          CONTAINER_CHECK_INTERVAL_SECS,
+                          check_container_state, duthost, container_name, True),
+                          "Failed to restart container '{}' after reset-failed was cleared".format(container_name))
+        else:
+            pytest.fail("Failed to restart container '{}'".format(container_name))
+
+    logger.info("Container '{}' was restarted".format(container_name))
+
+
+def verify_no_autorestart_with_non_critical_process(duthost, container_name, program_name,
+                                                    program_status, program_pid):
+    """
+    @summary: Kill a non-critical process in a container to verify whether the container
+              remains in the running state
+    """
+    if program_status == "RUNNING":
+        kill_process_by_pid(duthost, container_name, program_name, program_pid)
+    elif program_status in ["EXITED", "STOPPED", "STARTING"]:
+        pytest.fail("Program '{}' in container '{}' is in the {} state, expected 'RUNNING'"
+                    .format(program_name, container_name, program_status))
+    else:
+        pytest.fail("Failed to find program '{}' in container '{}'"
+                    .format(program_name, container_name))
+
+    logger.info("Checking whether container '{}' is still running...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS,
+                  CONTAINER_CHECK_INTERVAL_SECS,
+                  check_container_state, duthost, container_name, True),
+                  "Container '{}' was stopped unexpectedly".format(container_name))
+    logger.info("Container '{}' is running".format(container_name))
```

Comment:
Reword:
```
logger.info("Container '{}' did not stop".format(container_name))
```

Also, suggest restarting the program that you killed so that the container is in a fully healthy state before returning from this function using `supervisorctl start <program_name>`.

Reply:
Good suggestion. I will update. In this pytest script, first the non-critical process (`rsyslogd`) is killed and then test whether the container is stopped and restarted or not. Then a critical one is killed to see whether a container is restarted correctly. If the container is restarted, it will be in fully healthy state.